flink-runtime/src/main/java/org/apache/flink/runtime/jobmanager/web/JobManagerInfoServlet.java

```diff
@@ -220,6 +222,30 @@ else if("cancel".equals(req.getParameter("get"))) {
 
 				Await.ready(response, timeout);
 			}
+			else if ("stop".equals(req.getParameter("get"))) {
+				String jobId = req.getParameter("job");
+
+				response = jobmanager.ask(
+						new StopJob(JobID.fromHexString(jobId)),
+						timeout);
+
+				Await.ready(response, timeout);
+
+				String status = "success";
+				String cause = "";
+
+				Object rc = response.value().get().get();
```

Comment:
We could use a `Await.result` if you're interested in the return value of the `Future`.